packages/runtime/src/useCases/transport/devices/SetCommunicationLanguage.ts

```diff
@@ -0,0 +1,25 @@
+import { Result } from "@js-soft/ts-utils";
+import { LanguageISO639 } from "@nmshd/content";
+import { DeviceController } from "@nmshd/transport";
+import { Inject } from "typescript-ioc";
+import { RuntimeErrors, UseCase } from "../../common";
+
+export interface SetCommunicationLanguageRequest {
+    communicationLanguage: string;
```

Comment:
Is there a way to further restrict the type of the property `communicationLanguage` here so that only values that meet the ISO 639-1 standard can be used? Perhaps this would have advantages over performing the check according to the ISO 639-1 standard within the function call when the use case is exposed. A `SchemaValidator` could be added then too.

Reply:
that's actually a great Idea, we could yeet the manual check then!

Reply:
I'm generally not sure how the connector would handle enums properly, although this doesn't apply here - also is the policy for the schema validator to try adding it and removing it if things crash?

Reply:
the connector does not handle an enum at all. for the connector this is just a string.